librz/core/cmd_debug.c

```diff
@@ -3880,15 +3880,15 @@ static void rz_core_debug_esil(RzCore *core, const char *input) {
 		if (rz_debug_esil_watch_empty(core->dbg)) {
 			eprintf("Error: no esil watchpoints defined\n");
 		} else {
-			rz_core_cmd0(core, "aei");
+			rz_core_analysis_esil_init(core);
```

Comment:
Aren't you supposed to also run `aeim`/`cmd_esil_mem` with `aei`? `rz_core_analysis_esil_init` doesn't call it

Reply:
Also, calling rz_core_analysis_esil_init for each step seems redundant, maybe check if `core->analysis->esil` is initialized?

Reply:
I think `aei` command is not calling `aeim` as well.

Reply:
I preserved the logic as is. If you think it could be changed it should probably done in a separate PR. But usually yes, you need to run also the `aeim` too. Not sure why it wasn't done here before.